src/crypto/DeviceList.js

```diff
@@ -371,8 +512,11 @@ export default class DeviceList {
             }
         });
 
-        const finished = (success) => {
+        const finished = (newDevices) => {
             users.forEach((u) => {
+                this._devices[u] = newDevices[u];
+                this._dirty = true;
+
```

Comment:
The serialiser used to save the device keys to localstorage as they arrived, but since it store now lives in the `DeviceList` class, it made more sense to pass all the changes back in the promise.